src/lib/tls/tls_client.cpp

```diff
@@ -64,7 +64,10 @@ Client::Client(const Callbacks& callbacks,
 
 Handshake_State* Client::new_handshake_state(Handshake_IO* io)
    {
-   return new Client_Handshake_State(io, get_callbacks().handshake_msg());
+      return new Client_Handshake_State(io, 
+                                        std::bind(&TLS::Callbacks::handshake_msg, 
```

Comment:
We probably don't want to keep this bind here in the future. It would be a lot more readable and probably also more performant to just pass a Reference to TLS::Callbacks here.